pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -138,3 +123,71 @@ func TestReaderPool_ShouldCloseIdleLazyReaders(t *testing.T) {
 	require.Equal(t, float64(2), promtestutil.ToFloat64(metrics.lazyReader.loadCount))
 	require.Equal(t, float64(2), promtestutil.ToFloat64(metrics.lazyReader.unloadCount))
 }
+
+func TestReaderPool_PersistLazyLoadedBlock(t *testing.T) {
+	const idleTimeout = time.Second
+	ctx, tmpDir, bkt, blockID, metrics := prepareReaderPool(t)
+	defer func() { require.NoError(t, os.RemoveAll(tmpDir)) }()
+	defer func() { require.NoError(t, bkt.Close()) }()
```

Comment:
Can this also be moved into `prepareReaderPool`? Maybe something like `t.Cleanup(func() { require.NoError(t, bkt.Close()) }`?